pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -547,9 +551,17 @@ func fitsResourceQuota(fName kueue.ResourceFlavorReference, rName corev1.Resourc
 		// ClusterQueue are preempted.
 		mode = Preempt
 	}
+	if cq.Preemption.BorrowWithinCohort != nil && cq.Preemption.BorrowWithinCohort.Policy != kueue.BorrowWithinCohortPolicyNever {
+		// when preemption with borrowing is enabled, we can succeeded admitting the
+		// workload if preemption is used.
+		if rQuota.BorrowingLimit != nil && val <= rQuota.Nominal+*rQuota.BorrowingLimit {
```

Comment:
Indeed, additional check might be needed if the quota available for cohort < borrowingLimit. Added the check along with unit test. I guess we could validate that `cohortAvailableQuota >= borrowingLimit` earlier, but it is tricky to do in a `clusterqueue_webhook`, because it requires lookup to other clusterqueues.